staging/src/k8s.io/client-go/applyconfigurations/core/v1/persistentvolumeclaim.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by applyconfiguration-gen. DO NOT EDIT.
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	v1 "k8s.io/client-go/applyconfigurations/meta/v1"
+)
+
+// PersistentVolumeClaimApplyConfiguration represents an declarative configuration of the PersistentVolumeClaim type for use
+// with apply.
+type PersistentVolumeClaimApplyConfiguration struct {
+	v1.TypeMetaApplyConfiguration   `json:",inline"`
+	v1.ObjectMetaApplyConfiguration `json:"metadata,omitempty"`
+	Spec                            *PersistentVolumeClaimSpecApplyConfiguration   `json:"spec,omitempty"`
+	Status                          *PersistentVolumeClaimStatusApplyConfiguration `json:"status,omitempty"`
+}
+
+// PersistentVolumeClaimApplyConfiguration constructs an declarative configuration of the PersistentVolumeClaim type for use with
+// apply.
+func PersistentVolumeClaim(name, namespace string) *PersistentVolumeClaimApplyConfiguration {
+	b := &PersistentVolumeClaimApplyConfiguration{}
+	b.SetName(name)
+	b.SetNamespace(namespace)
+	b.SetKind("PersistentVolumeClaim")
+	b.SetAPIVersion("v1")
+	return b
+}
+
+// SetKind sets the Kind field in the declarative configuration to the given value.
+func (b *PersistentVolumeClaimApplyConfiguration) SetKind(value string) *PersistentVolumeClaimApplyConfiguration {
```

Comment:
Since we're allowing direct access to the underlying structs, we can't entirely prevent it. It gets automatically set by the constructors to the right values. Worth restricting on the setter?